### ISSUE TITLE
v4.0.0 and libfabric-devel-1.5.3-1.el7.x86_64 on Intel OPA1 systems don't mix

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#6075]
Turns out that the current OFI MTL - which in the v4.0.0 has higher priority than the PSM2 MTL - gets confused if built/run against the libfabric distributed in  libfabric-devel-1.5.3-1.el7.x86_64, resulting in flame out of things like IMB.  

```
[gr0540.localdomain:180658] mca:base:select: Auto-selecting mtl components
[gr0540.localdomain:180658] mca:base:select:(  mtl) Querying component [ofi]
[gr0540.localdomain:180658] mca:base:select:(  mtl) Query of component [ofi] set priority to 25
[gr0540.localdomain:180658] mca:base:select:(  mtl) Querying component [psm2]
[gr0540.localdomain:180658] mca:base:select:(  mtl) Query of component [psm2] set priority to 10
[gr0540.localdomain:180658] mca:base:select:(  mtl) Selected component [ofi]
[gr0540.localdomain:180658] mca: base: close: component psm2 closed
[gr0540.localdomain:180658] mca: base: close: unloading component psm2
[gr0540.localdomain:180658] select: initializing mtl component ofi
[gr0540.localdomain:180659] mca:base:select: Auto-selecting mtl components
[gr0540.localdomain:180659] mca:base:select:(  mtl) Querying component [ofi]
[gr0540.localdomain:180659] mca:base:select:(  mtl) Query of component [ofi] set priority to 25
[gr0540.localdomain:180659] mca:base:select:(  mtl) Querying component [psm2]
[gr0540.localdomain:180659] mca:base:select:(  mtl) Query of component [psm2] set priority to 10
[gr0540.localdomain:180659] mca:base:select:(  mtl) Selected component [ofi]
[gr0540.localdomain:180659] mca: base: close: component psm2 closed
[gr0540.localdomain:180659] mca: base: close: unloading component psm2
[gr0540.localdomain:180659] select: initializing mtl component ofi
[gr0540.localdomain:180658] mtl_ofi_component.c:315: mtl:ofi:provider_include = "(null)"
[gr0540.localdomain:180658] mtl_ofi_component.c:318: mtl:ofi:provider_exclude = "shm,sockets,tcp,udp,rstream"
[gr0540.localdomain:180658] mtl_ofi_component.c:347: mtl:ofi:prov: psm2
[gr0540.localdomain:180659] mtl_ofi_component.c:315: mtl:ofi:provider_include = "(null)"
[gr0540.localdomain:180659] mtl_ofi_component.c:318: mtl:ofi:provider_exclude = "shm,sockets,tcp,udp,rstream"
[gr0540.localdomain:180659] mtl_ofi_component.c:347: mtl:ofi:prov: psm2
[gr0540.localdomain:180658] select: init returned success
[gr0540.localdomain:180658] select: component ofi selected
[gr0540.localdomain:180659] select: init returned success
[gr0540.localdomain:180659] select: component ofi selected
[gr0540.localdomain:180658] mtl_ofi.h:493: fi_tsendddata failed: Function not implemented(-38)
[gr0540:180658] *** An error occurred in MPI_Bcast
[gr0540:180658] *** reported by process [1042219019,0]
[gr0540:180658] *** on communicator MPI_COMM_WORLD
[gr0540:180658] *** MPI_ERR_OTHER: known error not in list
[gr0540:180658] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[gr0540:180658] ***    and potentially your MPI job)
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
slurmstepd: error: *** STEP 15023647.11 ON gr0540 CANCELLED AT 2018-11-13T04:40:30 ***
srun: error: gr0540: task 0: Exited with exit code 16
srun: error: gr0540: task 1: Killed
```
simple workaround is to disable the ofi mtl via the --mca option on the mpirun command line, in the mca params conf file, or OMPI_MCA_mtl=^ofi

**Comment by matcabral** [comment]
@hppritcha, I see two things here: 
1) fi_tsendddata not implemented in ofi 1.5.3 psm2 provider, I will look into that 
2) MTL priorities, are you running with all local ranks? the psm2 mtl has a logic to lower its priority in this situation to favor vader. However, we still need to add that logic into ofi mtl. 
thanks,

**Comment by matcabral** [comment]
1) Seems to be a bug in the 1.5.3 OFI PSM2 provider that reports cq_data support when some tagged APIs did not support it. This is fixed in 1.6.0. 
2) I will create a PR for it.

**Comment by matcabral** [comment]
Hi @jsquyres , not yest. For issue 2) above #6082 was rejected (reverted in #6142). Therefore, I will open a new PR to remove the runtime priority changing in the PSM2 MTL.: 
https://github.com/open-mpi/ompi/blob/master/ompi/mca/mtl/psm2/mtl_psm2_component.c#L253

**Comment by matcabral** [comment]
Hi @hppritcha, I would like to close this issue based on the fact that the above described problem 1) is a bug in libfabric and that 2) was fixed in #6153 . feedback ?
